bin/test_access

```diff
@@ -4,22 +4,48 @@ source bin/stack_functions
 
 preamble_setup
 
+export CONTROLLER_NAME=127.0.0.1
+
 PROM_PT=8001
 
 function test_access {
-    desc=access-$1
-    echo Starting $desc test... | tee -a $TEST_RESULTS
+    adesc=access-$1
+    echo Starting $adesc test... | tee -a $TEST_RESULTS
     wget localhost:$PROM_PT -O- 2>/dev/null | grep 'nz-kiwi-t2sw1' | grep -oF '9a:02:57:1e:8f:01' | tee -a $TEST_RESULTS
     wget localhost:$PROM_PT -O- 2>/dev/null | grep 'nz-kiwi-t2sw2' | grep -oF '9a:02:57:1e:8f:02' | tee -a $TEST_RESULTS
 }
 
+function test_config {
+    fdesc=config-$1
+    fout_dir=$OUT_DIR/$fdesc
+    mkdir -p $fout_dir
+
+    echo Starting $fdesc tests | tee -a $TEST_RESULTS
+    echo $fout_dir
+    fetch_forch sys_config
+    api_result=$fout_dir/sys_config.json
+    jq '.faucet.dps."nz-kiwi-t2sw1".interfaces."1".native_vlan' $api_result | tee -a $TEST_RESULTS
+    jq '.faucet.dps."nz-kiwi-t2sw2".interfaces."1".acls_in[0]' $api_result | tee -a $TEST_RESULTS
+    jq '.faucet.dps."nz-kiwi-t2sw3".interfaces."1".native_vlan' $api_result | tee -a $TEST_RESULTS
+}
+
 echo %%% initial | tee -a $TEST_RESULTS
 test_access initial
 
-echo %%% l2-learning | tee -a $TEST_RESULTS
+# device are on default VLAN where only DHCP discover/request can go through
+echo %%% minimal-learning | tee -a $TEST_RESULTS
 docker exec forch-faux-1 arp -s 192.168.1.254 02:00:00:00:00:00
 docker exec forch-faux-2 arp -s 192.168.1.254 02:00:00:00:00:00
 docker exec forch-faux-1 bash -c "echo -n 'abc' > /dev/udp/192.168.1.254/67"
 docker exec forch-faux-2 bash -c "echo -n 'abc' > /dev/udp/192.168.1.254/68"
 sleep 1
-test_access l2-learning
+test_access minimal-learning
+
+restart_forch
```

Comment:
why is this necessary?

Reply:
So basically I wanted to test if forch can generate the behavioral config correctly and the traffic behaves correctly after faucet reloading new config, just to simulate the real case.

Reply:
But why restart it?   Shouldn't it be dynamically applying updates as the state of the system changes?  I would have exected forch to be started at the beginning of the sequence, and then the tests would be monitoring for the changes as they are dynamically applies.  This is, e.g., how the stack tests work with forch -- there it's just testing around links going up and down... ?

Reply:
> why restart it?
Actually that line starts forch for the first time, although it is called "restart_forch"

> I would have exected forch to be started at the beginning of the sequence, and then the tests would be monitoring for the changes as they are dynamically applies.
I Agree. The problem was that in this test setup forch is configured to load placements and behaviors upon starting. If starting forch at the very beginning of the test, the minimal-learning-acl will be overwritten, and the miminal-learning-vlan would not be tested. If instead we do not load placements and behaviors, it is not possible to load it later.

I guess we can either 1) monitoring placements.yaml and behaviors.yaml changes i Forch code and auto reload them, or 2) modify behaviors.yaml such that faux-1 and faux-2 stay on the default VLAN while faux-3 is assigned with a new VLAN so that we can test both minimal-learning and forch behavior?